src/configuration.rs

```diff
@@ -54,34 +132,193 @@ pub enum WhenConditionOperator {
+                [value].to_vec(),
+            )),
+            WhenConditionOperator::Matches => Ok(Expression::FunctionCall(
+                Expression::Ident(Arc::new("matches".to_string())).into(),
+                Some(Expression::Ident("attribute".to_string().into()).into()),
+                [value].to_vec(),
+            )),
         }
     }
 }
 
-#[derive(Deserialize, Debug, Clone)]
-#[serde(rename_all = "camelCase")]
-pub struct PatternExpression {
-    pub selector: String,
-    pub operator: WhenConditionOperator,
-    pub value: String,
+pub fn type_of(path: &str) -> ValueType {
+    match path {
+        "request.time" => ValueType::Timestamp,
+        "request.id" => ValueType::String,
+        "request.protocol" => ValueType::String,
+        "request.scheme" => ValueType::String,
+        "request.host" => ValueType::String,
+        "request.method" => ValueType::String,
+        "request.path" => ValueType::String,
+        "request.url_path" => ValueType::String,
+        "request.query" => ValueType::String,
+        "request.referer" => ValueType::String,
+        "request.useragent" => ValueType::String,
+        "request.body" => ValueType::String,
+        "source.address" => ValueType::String,
+        "source.service" => ValueType::String,
+        "source.principal" => ValueType::String,
+        "source.certificate" => ValueType::String,
+        "destination.address" => ValueType::String,
+        "destination.service" => ValueType::String,
+        "destination.principal" => ValueType::String,
+        "destination.certificate" => ValueType::String,
+        "connection.requested_server_name" => ValueType::String,
+        "connection.tls_session.sni" => ValueType::String,
+        "connection.tls_version" => ValueType::String,
+        "connection.subject_local_certificate" => ValueType::String,
+        "connection.subject_peer_certificate" => ValueType::String,
+        "connection.dns_san_local_certificate" => ValueType::String,
+        "connection.dns_san_peer_certificate" => ValueType::String,
+        "connection.uri_san_local_certificate" => ValueType::String,
+        "connection.uri_san_peer_certificate" => ValueType::String,
+        "connection.sha256_peer_certificate_digest" => ValueType::String,
+        "ratelimit.domain" => ValueType::String,
+        "request.size" => ValueType::Int,
+        "source.port" => ValueType::Int,
+        "destination.port" => ValueType::Int,
+        "connection.id" => ValueType::Int,
+        "ratelimit.hits_addend" => ValueType::Int,
+        "request.headers" => ValueType::Map,
+        "request.context_extensions" => ValueType::Map,
+        "source.labels" => ValueType::Map,
+        "destination.labels" => ValueType::Map,
+        "filter_state" => ValueType::Map,
+        "connection.mtls" => ValueType::Bool,
+        "request.raw_body" => ValueType::Bytes,
+        "auth.identity" => ValueType::Bytes,
+        _ => ValueType::Bytes,
```

Comment:
This probably is best to first try a `String` and fallback to `Bytes` if not a valid utf-8 sequence... 
That's mean that someone accessing a value "deeper" within a well-know attribute (e.g. `metadata.filter_metadata.envoy\.filters\.http\.ext_authz.identity.userid`), it try to parse it as a string first